lib/msf/core/post/file.rb

```diff
@@ -783,12 +783,9 @@ def _read_file_powershell_fragment(filename, chunk_size, offset = 0)
   # @param data [String] String to check for non-ansi or newline chars
   # @return bool
   def _can_echo?(data)
-    data.each_char do |char|
-      unless char.ascii_only? || char == '\n' || char == '"'
```

Comment:
Updated to use the new line character `"\n"` here, i.e:

```
2.7.2 :017 > '\n'.length
 => 2 
2.7.2 :018 > "\n".length
 => 1 
```

As well as fixing the the logic